internal/runbits/buildscript/merge.go

```diff
@@ -0,0 +1,58 @@
+package buildscript
+
+import (
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/locale"
+	"github.com/ActiveState/cli/pkg/localcommit"
+	"github.com/ActiveState/cli/pkg/platform/api/mono/mono_models"
+	"github.com/ActiveState/cli/pkg/platform/authentication"
+	"github.com/ActiveState/cli/pkg/platform/model"
+	"github.com/ActiveState/cli/pkg/platform/runtime/buildexpression/merge"
+	"github.com/ActiveState/cli/pkg/platform/runtime/buildscript"
+	"github.com/ActiveState/cli/pkg/project"
+	"github.com/go-openapi/strfmt"
+)
+
+// Merge merges the local build script with the remote buildexpression (not script) for a given
+// UUID, performing the given merge strategy (e.g. from model.MergeCommit).
+func Merge(proj *project.Project, remoteCommit strfmt.UUID, strategies *mono_models.MergeStrategies, auth *authentication.Auth) error {
+	// Verify we have a build script to merge.
+	script, err := buildscript.NewScriptFromProjectDir(proj.Dir())
+	if err != nil && !buildscript.IsDoesNotExistError(err) {
+		return errs.Wrap(err, "Could not get local build script")
+	}
+	if script == nil {
+		return nil // no build script to merge
+	}
```

Comment:
Maybe the user previously made a local modification, tried to pull, got a conflict that they did not want to resolve, and deleted the local build script in the hope that the remote one would be pulled in automatically (which it will eventually).

Reply:
I'd rather we handle this as an explicit use-case though. For now, I think it's appropriate for us to error out on this case.

I've been bit many times introducing "features" that weren't requested but I figured they were easy enough to facilitate while I was at it.

Reply:
What about the case where the user has an existing project that does not have a build script/build plan because this is the first time they are pulling after build plans were introduced? In that case, every initial pull would be an error.

I'm just thinking there are legitimate reasons why a build script may not exist, and so the default should be to accept the remote/incoming one; no merge is needed.

Reply:
That's a great use-case to consider, but luckily one that we already have covered with:

https://github.com/ActiveState/cli/blob/a2e9a48e6581ceda1ed08eff601cdba015cf0a9d/pkg/platform/runtime/buildscript/file.go#L36-L36

This is called from the pull runner as well as other runners. So the scenario you're covering here still should not happen and we should make it an error condition as a result, imo.

Reply:
I just realized this is only called at the end. So your scenario could still play out. I'm filing a story to find a better solution here as I don't want our code to have to be aware of this scenario at every turn if we can help it.

Reply:
Okay, done.